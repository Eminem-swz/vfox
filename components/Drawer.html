<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Drawer 抽屉 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.ebc52af7.css">
    <link rel="modulepreload" href="/vfox/assets/Home.2f3d3943.js">
    <link rel="modulepreload" href="/vfox/assets/app.e71fba3c.js">
    <link rel="modulepreload" href="/vfox/assets/components_Drawer.md.6beadd9b.lean.js">
    <link rel="modulepreload" href="/vfox/assets/app.e71fba3c.js">
    <meta name="twitter:title" content="Drawer 抽屉 | Vfox for Vue3">
    <meta property="og:title" content="Drawer 抽屉 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-7161a24b><div class="sidebar-button" data-v-7161a24b><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-7161a24b data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-7161a24b></div><div class="nav" data-v-7161a24b><nav class="nav-links" data-v-7161a24b data-v-15acbf05><!--[--><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/" data-v-641633f9>指南 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/design/" data-v-641633f9>设计 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item active" href="/vfox/components/Button" data-v-641633f9>组件 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/apis/LocalStorage" data-v-641633f9>API <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-dropdown-link" data-v-15acbf05 data-v-90f2c308><button class="button" data-v-90f2c308><span class="button-text" data-v-90f2c308>选择版本</span><span class="right button-arrow" data-v-90f2c308></span></button><ul class="dialog" data-v-90f2c308><!--[--><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>1.x</span><span class="icon" data-v-57d767ef><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-57d767ef><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item active" href="/vfox/" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>2.x</span><span class="icon" data-v-57d767ef><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-6b49cdcd><nav class="nav-links nav" data-v-6b49cdcd data-v-15acbf05><!--[--><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/" data-v-641633f9>指南 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/design/" data-v-641633f9>设计 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item active" href="/vfox/components/Button" data-v-641633f9>组件 <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-link" data-v-15acbf05 data-v-641633f9><a class="item" href="/vfox/apis/LocalStorage" data-v-641633f9>API <!----></a></div></div><div class="item" data-v-15acbf05><div class="nav-dropdown-link" data-v-15acbf05 data-v-90f2c308><button class="button" data-v-90f2c308><span class="button-text" data-v-90f2c308>选择版本</span><span class="right button-arrow" data-v-90f2c308></span></button><ul class="dialog" data-v-90f2c308><!--[--><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>1.x</span><span class="icon" data-v-57d767ef><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-57d767ef><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-90f2c308><div class="nav-dropdown-link-item" data-v-90f2c308 data-v-57d767ef><a class="item active" href="/vfox/" data-v-57d767ef><span class="arrow" data-v-57d767ef></span><span class="text" data-v-57d767ef>2.x</span><span class="icon" data-v-57d767ef><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-6b49cdcd><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Drawer">Drawer</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#events">Events</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#visiblestate-值说明">VisibleState 值说明</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/QrCode">QrCode</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><!-- TODO: make this button accessible --><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://cdn.fox2.cn/2.x/demo/#/Drawer" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="drawer-抽屉"><a class="header-anchor" href="#drawer-抽屉" aria-hidden="true">#</a> Drawer 抽屉</h1><p>用于在屏幕边缘显示应用导航等内容的面板。</p><h2 id="props"><a class="header-anchor" href="#props" aria-hidden="true">#</a> Props</h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>v-model:visible</td><td>boolean</td><td>false</td><td>否</td><td>是否显示</td></tr><tr><td>title</td><td>string</td><td></td><td>否</td><td>标题，设置标题后展示头部栏</td></tr><tr><td>placement</td><td>string</td><td>&#39;bottom&#39;</td><td>否</td><td>从哪展开，可选值：&#39;bottom&#39;, &#39;top&#39;, &#39;left&#39;, &#39;right&#39;</td></tr><tr><td>show-close</td><td>boolean</td><td>false</td><td>否</td><td>是否显示关闭按钮，显示按钮后展示头部栏</td></tr></tbody></table><h2 id="events"><a class="header-anchor" href="#events" aria-hidden="true">#</a> Events</h2><table><thead><tr><th>事件</th><th>描述</th><th>回调函数参数</th></tr></thead><tbody><tr><td>visible-state-change</td><td>展示隐藏时触发</td><td>{ state: <a href="./Drawer.html#visiblestate-值说明">VisibleState</a> }</td></tr></tbody></table><h3 id="visiblestate-值说明"><a class="header-anchor" href="#visiblestate-值说明" aria-hidden="true">#</a> VisibleState 值说明</h3><table><thead><tr><th>值</th><th>说明</th><th>备注</th></tr></thead><tbody><tr><td>show</td><td>展示时触发</td><td></td></tr><tr><td>shown</td><td>展示且动画结束后触发</td><td></td></tr><tr><td>hide</td><td>隐藏时触发</td><td>可能携带其他参数 cancel, maskClick, closeClick 等</td></tr><tr><td>hidden</td><td>隐藏且动画结束后触发</td><td>可能携带其他参数 cancel, maskClick, closeClick 等</td></tr></tbody></table><h2 id="slots"><a class="header-anchor" href="#slots" aria-hidden="true">#</a> Slots</h2><div class="language-"><pre><code>&lt;fx-drawer title=&quot;菜单&quot; placement=&quot;right&quot;&gt;
  &lt;fx-tab /&gt;
&lt;/fx-drawer&gt;
</code></pre></div></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/ScrollView" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>ScrollView</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/Modal" data-v-38ede35f><span class="text" data-v-38ede35f>Modal</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"index.md\":\"8bae8ee2\",\"apis_drawer.md\":\"73c4adeb\",\"apis_intersectionobserver.md\":\"3f62fc9d\",\"apis_localstorage.md\":\"119fc2c8\",\"apis_modal.md\":\"1a392ada\",\"apis_notify.md\":\"98864ea5\",\"apis_popover.md\":\"45e8de57\",\"apis_preview.md\":\"00077c6f\",\"apis_scroll.md\":\"1b6e1cc6\",\"apis_selectorquery.md\":\"13691696\",\"apis_toast.md\":\"54a2bffb\",\"apis_index.md\":\"0aa86774\",\"components_actionsheet.md\":\"dd9d9cee\",\"components_activityindicator.md\":\"d119fb2d\",\"components_avatar.md\":\"33d1fbdb\",\"components_backtop.md\":\"ec7298cd\",\"components_badge.md\":\"e5493e09\",\"components_button.md\":\"f97174cc\",\"components_calendar.md\":\"8d384e71\",\"components_cascader.md\":\"eecf2b13\",\"components_cell.md\":\"cedf5a22\",\"components_checkbox.md\":\"2a577e79\",\"components_circleprogress.md\":\"17a3d56e\",\"components_collapse.md\":\"e582d739\",\"components_copy.md\":\"f4ac98ed\",\"components_countdown.md\":\"632987e7\",\"components_countup.md\":\"dbf51c15\",\"components_datepicker.md\":\"a3e011ef\",\"components_dialog.md\":\"2609d5cc\",\"components_divider.md\":\"a2851e41\",\"components_drawer.md\":\"6beadd9b\",\"components_dropdown.md\":\"688dbf2a\",\"components_empty.md\":\"c09aef70\",\"components_fixed.md\":\"21f6d429\",\"components_flatlist.md\":\"1708e75d\",\"components_form.md\":\"764e1793\",\"components_group.md\":\"2142cc66\",\"components_icon.md\":\"223ddbe3\",\"components_image.md\":\"49b4c7f5\",\"components_imagepreview.md\":\"5d3701b8\",\"components_imageuploader.md\":\"6225f452\",\"components_indexview.md\":\"9d9058d2\",\"components_input.md\":\"8e96bcd7\",\"components_layout.md\":\"29d9b95b\",\"components_loadmore.md\":\"d63af18a\",\"components_modal.md\":\"1db6be51\",\"components_navbar.md\":\"9667ea53\",\"components_noticebar.md\":\"29ae091a\",\"components_notify.md\":\"dbc48d89\",\"components_numberkeyboard.md\":\"32333df3\",\"components_order.md\":\"82b6f83f\",\"components_pagination.md\":\"177244f5\",\"components_picker.md\":\"007487d3\",\"components_popdialog.md\":\"2c74aae6\",\"components_popmenu.md\":\"2bd6c6ec\",\"components_popover.md\":\"3ec7b53d\",\"components_price.md\":\"5fe81f64\",\"components_progress.md\":\"1154b336\",\"components_qrcode.md\":\"b5774bc5\",\"components_radio.md\":\"565f3a0b\",\"components_range.md\":\"d34036fe\",\"components_rate.md\":\"c681ffbc\",\"components_result.md\":\"dcaf0f44\",\"components_scrolltab.md\":\"ce7a7309\",\"components_scrollview.md\":\"72188c68\",\"components_searchbar.md\":\"35b61bbd\",\"components_sidetab.md\":\"3497e693\",\"components_skeleton.md\":\"f9c11593\",\"components_slider.md\":\"da41e9c5\",\"components_stepper.md\":\"05b318cd\",\"components_steps.md\":\"17100f31\",\"components_sticky.md\":\"f0062752\",\"components_stopwatch.md\":\"4d3a29dc\",\"components_swipecell.md\":\"f18e0ba3\",\"components_swiper.md\":\"e8b04d26\",\"components_switch.md\":\"1ae3e937\",\"components_tab.md\":\"0af0cfb0\",\"components_tabbar.md\":\"648ba82e\",\"components_tabview.md\":\"a126f8c9\",\"components_tag.md\":\"425528c0\",\"components_timeago.md\":\"1a67a7e4\",\"components_timeline.md\":\"247cce04\",\"components_toast.md\":\"530b1188\",\"components_index.md\":\"11a7db9b\",\"design_index.md\":\"0ab5cd15\"}")</script>
    <script type="module" async src="/vfox/assets/app.e71fba3c.js"></script>
  </body>
</html>