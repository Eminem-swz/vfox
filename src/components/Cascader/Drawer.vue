<template>
  <drawer ref="drawer" :title="title" align="right" :visible="true">
    <div ref="dropdown" :class="[prefix + '-cascader_groups']">
      <div
        :class="[prefix + '-cascader_group', prefix + '-vertical-hairline']"
        v-for="(list, listIndex) in dropdown"
        :key="listIndex"
      >
        <ul :class="[prefix + '-cascader_list']" :data-index="listIndex">
          <li
            :class="[
              prefix + '-cascader_item',
              prefix + '-horizontal-hairline',
              {
                selected: item.selected,
                disabled: item.disabled
              }
            ]"
            v-for="(item, index) in list"
            :key="item.value"
            :data-index="index"
            @click="onItemClick($event, item)"
          >
            <span :class="[prefix + '-cascader_item-text']">
              {{ item.label }}
            </span>
          </li>
        </ul>
      </div>
    </div>
  </drawer>
</template>

<script>
import Drawer from '../Drawer'
import { SDKKey } from '../../config'
import { array2String, getDefaultSelecteds, parseDropdownList } from './util'
import { frameTo } from '../../helpers/animation'

export default {
  name: SDKKey + '-cascader-picker',
  components: { Drawer },
  props: {
    options: {
      type: Array,
      default() {
        return []
      }
    },
    mode: String,
    separator: String,
    placeholder: String
  },
  data() {
    return {
      prefix: SDKKey,

      dropdown: [],

      labels: []
    }
  },
  computed: {
    title() {
      if (this.labels[0]) {
        return array2String(this.labels, this.mode, this.separator)
      }

      return this.placeholder || ' '
    }
  },
  watch: {},
  created() {},
  ready() {},
  mounted() {},
  updated() {},
  attached() {},
  methods: {
    show() {
      this.$refs.drawer && this.$refs.drawer.show()
    },

    hide() {
      this.$refs.drawer && this.$refs.drawer.hide()
    },

    destroy() {
      this.$refs.drawer && this.$refs.drawer.destroy()
    },

    parseDropdown(selecteds) {
      this.labels = []
      let optionList = parseDropdownList(this.mode, 0, this.options)

      if (selecteds.length === 0) {
        // 如果没有默认值，尝试获取默认值
        selecteds = getDefaultSelecteds(this.mode)
      }

      const menuGroup = []
      const menuLabels = []
      const menuValues = []
      //   let lastGroupSelected = false

      for (let i = 0; i <= selecteds.length; i++) {
        let selected = selecteds[i]
        const menuList = []
        let nextParent = null
        // lastGroupSelected = false

        if (optionList) {
          for (let j = 0; j < optionList.length; j++) {
            let optionItem = optionList[j]

            let menuItem = {
              label: optionItem.label,
              value: optionItem.value,
              selected: false,
              hasChildren: optionItem.hasChildren,
              disabled: optionItem.disabled ? true : false
            }

            menuItem.values = menuValues.concat(menuItem.value)
            menuItem.labels = menuLabels.concat(menuItem.label)

            if (selected != null && menuItem.value === selected) {
              // 找到
              menuItem.selected = true
              //   lastGroupSelected = true

              this.labels.push(menuItem.label)

              if (menuItem.hasChildren) {
                nextParent = optionItem
              }
            }

            menuList.push(menuItem)
          }
          menuGroup.push(menuList)
        }

        if (!nextParent) {
          break
        } else {
          optionList = parseDropdownList(this.mode, i + 1, nextParent)
          menuValues.push(nextParent.value)
          menuLabels.push(nextParent.label)
        }
      }

      this.dropdown = menuGroup

      this.$nextTick(() => {
        this.afterUpdate()
      })
    },

    afterUpdate() {
      const $dropdown = this.$refs.dropdown
      const $lists = $dropdown.querySelectorAll(`.${SDKKey}-cascader_list`)
      const $selecteds = $dropdown.querySelectorAll('.selected')

      if ($lists[1]) {
        let $next
        if ($lists.length > $selecteds.length) {
          // 还有未选择
          $next = $lists[$selecteds.length]
        } else {
          // 选择完毕
          $next = $lists[$lists.length - 1]
        }

        $next = $next.parentNode
        const $groups = $next.parentNode

        let to

        if ($next.offsetWidth >= document.documentElement.offsetWidth * 0.9) {
          to = $next.offsetLeft
        } else {
          to = $next.offsetLeft - ($groups.offsetWidth - $next.offsetWidth)
        }

        if (to >= 0) {
          frameTo({
            from: $groups.scrollLeft,
            to,
            duration: 100,
            progress(res) {
              $groups.scrollLeft = res.current
            }
          })
        }

        // $next.scrollIntoView(
        //   $next.offsetWidth < document.documentElement.offsetWidth
        // )
      } else {
        $dropdown.scrollLeft = 0
      }

      $lists.forEach(($list, index) => {
        if ($selecteds[index]) {
          frameTo({
            from: $list.scrollTop,
            to: $selecteds[index].offsetTop,
            duration: 100,
            progress(res) {
              $list.scrollTop = res.current
            }
          })
        } else {
          $list.scrollTop = 0
        }
      })
    },

    /**
     * 选项点击
     */
    onItemClick(e, item) {
      if (item.disabled) {
        return
      }

      if (item.hasChildren) {
        this.parseDropdown(item.values)
      } else {
        this.$parent.onSelect(item)
      }
    }
  }
}
</script>
