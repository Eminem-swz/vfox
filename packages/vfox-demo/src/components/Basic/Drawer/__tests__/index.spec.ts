import { shallowMount } from '@vue/test-utils'
import { Drawer } from '@/Drawer'

describe('Drawer', () => {
  test('should render prop title="Drawer" correctly', () => {
    const wrapper = shallowMount(Drawer, {
      props: {
        title: 'Drawer'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop showClose=true correctly', () => {
    const wrapper = shallowMount(Drawer, {
      props: {
        showClose: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render prop placement="left" correctly', () => {
    const wrapper = shallowMount(Drawer, {
      props: {
        placement: 'left'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
