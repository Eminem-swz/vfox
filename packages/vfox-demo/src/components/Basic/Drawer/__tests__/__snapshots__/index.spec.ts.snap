// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render prop placement="left" correctly 1`] = `<anonymous-stub to="body"></anonymous-stub>`;

exports[`Drawer should render prop showClose=true correctly 1`] = `<anonymous-stub to="body"></anonymous-stub>`;

exports[`Drawer should render prop title="Drawer" correctly 1`] = `<anonymous-stub to="body"></anonymous-stub>`;
